doc/porting_guide.md

```diff
@@ -3,343 +3,366 @@
 
 ##### Preprocessor Definitions
 
-Executing a makefile build on OSX or Linux/Unix with your desired config flags (see below) and then examining the make output is highly recommended.  By calling make with attribute 'MD=' you can reveal compiler and linker parameters, showing you which Preprocessor Definitions need to be migrated to your IDE environment.
+Executing a makefile build on OSX or Linux/Unix with your desired config flags (see below) and then examining the make output is highly recommended.  
+
+Additionally, by calling make with attribute 'MD=' you can reveal compiler and linker parameters, showing you which Preprocessor Definitions need to be migrated to your IDE environment.
 
 Importing these definitions in an IDE varies considerably by IDE.  Some IDEs let you define these in a source or configuration file. Others require you to enter them as fields in a project configuration window.
 
 ##### C Sourcefiles
 
-A vast majority of the sources in the Xively Client Library are applicable to all platforms, some source files might not be needed depending on your configuration choices.  Once again, like the preprocessor flags, we recommend running a make build with your desired configuration to determine which source files need to be added to your IDE builds, and which includes paths should be set.
+A vast majority of the sources in the `src/libxively` directory Xively Client Library are applicable to all platforms, with the platform sepecific files residing in the `src/bsp` directory.
+
+However some source files might not be needed depending on your configuration choices.  Once again, like the preprocessor flags, we recommend running a make build with your desired configuration to determine which source files need to be added to your IDE builds, and which includes paths should be set.
 
 #### First Build on OSX or Linux/Unix
 
 Firing a bare `make` should be enough for an OSX build. It defaults to *posix* platform and to wolfSSL for a TLS solution. It also includes the download (from github repo) and build of the wolfSSL library.
 
 Run command
 
-`make`
+	make
 
 Alternatively one can use mbedTLS instead of wolfSSL if preferred:
 
-`make XI_BSP_TLS=mbedTLS`
+	make XI_BSP_TLS=mbedTLS
 
-Occasionally cleaning the build directories are necessary. This is done with the commands
+Occasionally cleaning the build directories are necessary. This is done with either of the following commands:
 
-`make clean`
-    OR
-`make clean_all`
+	make clean
+	make clean_all
 
-##### Building the examples
+##### Building the Examples
+From the base directory of the repository:
 
-`cd examples`
+	cd examples
+	make
 
-`make`
+Here if the library was previously built with the mbedTLS setting then the XI_BSP_TLS variable is necessary.  Otherwise you will get link errors.
 
-here if previously library was built with mbedTLS setting the XI_BSP_TLS variable is necessary to tell the linker which TLS lib to link agains:
+	make XI_BSP_TLS=mbedTLS
```

Comment:
unfortunately here all lower case is required since this will be inserted into the template file name: mt-tls-$(XI_BSP_TLS).mk . And the filename is stilll lowercase :(